src/server/game/AI/CreatureAI.cpp

```diff
@@ -276,6 +280,7 @@ void CreatureAI::EngagementStart(Unit* who)
     }
     _isEngaged = true;
 
+    me->InterruptNonMeleeSpells(false);
```

Comment:
Some creatures must continue channeling after aggro

Reply:
Interrupting the channeling is the rule, continue is the exception, I cannot remember a creature or boss that continue channeling once aggro.

Reply:
Isn't there a attribute to don't interrupt channeling on aggro?
I'm 100% sure it is invalid for master but I can't remember any wotlk creature with exception. Cmangos does it manually in each script even in tbc branch

Maybe it affects Magtheridon now?